third_party/move/move-compiler-v2/src/file_format_generator/function_generator.rs

```diff
@@ -986,11 +986,13 @@ impl<'env> FunctionContext<'env> {
 
 impl<'env> BytecodeContext<'env> {
     /// Determine whether the temporary is alive (used) in the reachable code after this point.
-    /// If `dest_check` is true, temporary should not be written to by the current instruction
-    /// for it to be considered alive.
-    pub fn is_alive_after(&self, temp: TempIndex, dest_check: bool) -> bool {
+    /// If `is_source` is true, then we are checking for liveness of `temp` in the context of it
```

Comment:
This is even more confusing.  The old one at least made operational sense.  This one seems just wrong.

Reply:
maybe rename it to `temp_is_rewritten` and then we can remove this comment because we will explain how it is used below?

Reply:
> This one seems just wrong.

When we have `t = foo(t)`, and we are generating code for the RHS, we want to check if `t` is alive for the "source" `t` (for example, when pushing stuff onto the stack to call `foo`). After the whole operation is done, we may want to check if `t` is alive (eg., at the end of the basic block) in the non-source context.

It would be helpful to understand why this seems "just wrong" or is even more confusing. Naming is subjective, and it might not be obvious to others why a subjective statement holds, so when a review request is made, a bit more elaboration goes a long way. Otherwise, there could be multiple unnecessary back-and-forths.

> maybe rename it to `temp_is_rewritten` and then we can remove this comment because we will explain how it is used below?

It is unclear to me why this renaming is better than say `dest_check`. I like when parameters are explained in the function doc rather than expecting the reader to dig through the implementation to find what it is used for.

Summary: I will go back to the original "operationally named" `dest_check` to move forward. If you have a convincing name, do let me know and I will try to accommodate it.

Reply:
On second thought, maybe it's not "just wrong" but is more confusing.

- `self.is_alive_after(temp, false)` = value in`temp` after this instruction is alive.

- `self.is_alive_after(temp, true)` = value of source `temp` is alive after reading sources of this instruction (before writing destination; if idestination is `temp` then value is killed).

The second case is really just testing for simple liveness, but at a code point within the instruction.  I think that's what makes it confusing.  It's no longer really `is_alive_after` but is now `is_alive_after_reading_sources`.

Reply:
Thank you for the elaboration. I agree with your assessment of what is happening.

How about `self.is_alive_after(temp, consuming_sources: true)` (this is how it would look in my IDE), that is, use the name `consuming_sources` for the boolean flag. I use consuming because of the move semantics (and the decisions, such as to use moveloc vs. copyloc would be based on this flag).

Reply:
I honestly don't think it works to elaborate the name of the function with the name of the parameter.  Both cases "consume sources" so having that be the name of the parameter isn't great.

Better if the parameter stands alone.  So going back to the original parameter name and just adding a little more comment would be better.

Reply:
Alrighty.